job/job.go

```diff
@@ -94,6 +126,11 @@ type Metadata struct {
 	LastAttemptedRun time.Time `json:"last_attempted_run"`
 }
 
+type Header struct {
```

Comment:
Is there a reason you choose to use a custom type rather than `http.Header`? https://golang.org/pkg/net/http/#Header